From 59fb62770b6a682cb30e1c20b39ee447630a561b Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Mon, 17 Jul 2023 18:20:51 +0800
Subject: [PATCH] update the hostkeys of ssh session connection

The version of libssh has been updated, the old hostkeys are not
supported on new libssh. Netconf client could not connected to the
server.

Signed-off-by: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
---
 src/ssh.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/ssh.c b/src/ssh.c
index 06a9b775..b747eed8 100644
--- a/src/ssh.c
+++ b/src/ssh.c
@@ -223,8 +223,7 @@ struct nc_session *nc_session_connect_libssh_socket(const char* username, const
 	ssh_options_set(retval->ssh_sess, SSH_OPTIONS_FD, &retval->transport_socket);
 	ssh_options_set(retval->ssh_sess, SSH_OPTIONS_TIMEOUT, &timeout);
     if (ssh_options_set(retval->ssh_sess, SSH_OPTIONS_HOSTKEYS,
-                        "ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,"
-                        "ecdsa-sha2-nistp256,ssh-rsa,ssh-dss,ssh-rsa1")) {
+                        "rsa-sha2-512,rsa-sha2-256")) {
         /* ecdsa is probably not supported... */
         ssh_options_set(retval->ssh_sess, SSH_OPTIONS_HOSTKEYS, "ssh-ed25519,ssh-rsa,ssh-dss,ssh-rsa1");
     }
-- 
2.25.1

